lib/instruments.js

```diff
@@ -351,23 +351,28 @@ class Instruments {
 
   /* PROCESS MANAGEMENT */
   async shutdown () {
-    if (this.proc) {
-      log.debug('Starting shutdown.');
-      let wasTerminated = false;
-      // monitoring process termination
-      let termDelay = cancellableDelay(this.termTimeout);
-      let termPromise = termDelay.catch(B.CancellationError, () => {});
-      this.setExitListener(() => {
-        this.proc = null;
-        wasTerminated = true;
-        termDelay.cancel();
-        this.onShutdownDeferred.resolve();
-      });
-      log.debug('Sending sigterm to instruments');
-      this.proc.kill('SIGTERM');
-      await termPromise;
-      if (!wasTerminated) {
-        throw new Error(`Instruments did not terminate after ${this.termTimeout / 1000} seconds!`);
+    if (!this.proc) return;
+    
+    log.debug('Starting shutdown.');
+    let wasTerminated = false;
+    // monitoring process termination
+    let termDelay = cancellableDelay(this.termTimeout);
+    let termPromise = termDelay.catch(B.CancellationError, () => {});
+    this.setExitListener(() => {
+      this.proc = null;
+      wasTerminated = true;
+      termDelay.cancel();
+      this.onShutdownDeferred.resolve();
+    });
+    log.debug('Sending sigterm to instruments');
+    this.proc.kill('SIGTERM');
+    await termPromise;
+    if (!wasTerminated) {
+      log.warn(`Instruments did not terminate after ${this.termTimeout / 1000} seconds!`);
+      log.debug('Sending SIGKILL');
+      this.proc.kill('SIGKILL');
+      if (_.isFunction(this.exitListener)) {
+        this.exitListener();
```

Comment:
Pretty much. The behavior now is to throw an error, but it is just logged at the driver level. So this is just a last-ditch effort to kill it. If it doesn't work, we've logged that fact and continue as would have previously.